### PR TITLE
fix: numpy returns handles more name, type cases

### PR DESCRIPTION
**Pull request by machow**
Addresses #173 . I'm not particularly good and regexes, so definitely feel free to take this as more of a demonstration of the issue. Happy to make any tweaks / changes :)

### CODE REVIEW

File: src/griffe/docstrings/numpy.py
Comment (pawamoy) [inline src/griffe/docstrings/numpy.py:222]: Not tested, but this would look better to me:

```suggestion
        (?::\s*)?(?P<type>{_RE_TYPE})\s*  # just type
        |  # or
        :\s*  # no name, no type
```

The difference is: we don't allow leading spaces before the colon, but allow trailing spaces after it.

File: tests/test_docstrings/test_numpy.py
Comment (pawamoy) [inline tests/test_docstrings/test_numpy.py:378]: Shouldn't it pick the name here?

```suggestion
        DocstringReturn(name="x", annotation=None, description="Name only"),
```

**Review by pawamoy — COMMENTED** [review]
Thanks @machow for all your contributions, you're awesome :smile: 
A few comments :slightly_smiling_face:

File: src/griffe/docstrings/numpy.py
Comment (machow) [reply src/griffe/docstrings/numpy.py:222]: Did a bit of tinkering just now, and I think because the `_RE_TYPE` regex is `.+`, the "just type" regex ends up capturing the "no name, no type" cases.

I think we may need the "no name, no type" regex first, since it's a subset of things our "just type" captures.

File: tests/test_docstrings/test_numpy.py
Comment (machow) [reply tests/test_docstrings/test_numpy.py:378]: Ahhhh, assert_element_equal doesn't test name 😬. Going through now to see how to work it in

File: src/griffe/docstrings/numpy.py
Comment (pawamoy) [reply src/griffe/docstrings/numpy.py]: Ha, sorry for making you do regex gymnastics! I thought my suggestion covered it, but if it's easier to have the  "nothing" case before the "just type" one, then please keep them in that order! I prefer the regex to stay simple and not individually cover badly-supported cases from composed ones. Each regex deals with its own issues :slightly_smiling_face:

**Comment by machow** [comment]
@pawamoy, alright, I think assert_element_equal should be checking all the attributes now. Happy to make any other useful changes

**Comment by pawamoy** [comment]
I'm waiting for an answer at https://github.com/numpy/numpydoc/issues/463. Numpydoc maintainers don't particularly like the `:` and `: int` syntax. If we can't find a better alternative, we'll go ahead with it :slightly_smiling_face:

**Comment by pawamoy** [comment]
Lets wait a few more days, and then proceed to merge. I don't think the discussion on numpydoc will yield immediate results anyway. I'm OK with driving changes rather than waiting for them :slightly_smiling_face:

**Review by pawamoy — APPROVED** [review]
OK, LGTM now, what do you think? I'm ready to squash and merge :slightly_smiling_face:

**Comment by pawamoy** [comment]
Merging, thanks again @machow !

**Pull request merged**
